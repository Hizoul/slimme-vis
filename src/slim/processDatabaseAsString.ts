
export interface IClass {
  label: string
  values: number
}

export interface Database {[index: string]: number[][] | undefined}

const processDatabaseAsString = (databaseContent: string) => {
  const classes: Database = {}
  for (const line of databaseContent.split("\n")) {
    const values = line.split(" ").map((value) => {
      if (value.length > 0) {
        return Number(value)
      }
    }).filter((val) => val != null)
    const classKey = String(values[0])
    if (classes[classKey] == null) {
      classes[classKey] = []
    }
    classes[classKey].push(values.splice(1))
  }
  return classes
}

export default processDatabaseAsString
