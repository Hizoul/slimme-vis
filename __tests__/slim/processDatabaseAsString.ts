import { sha512 } from "js-sha512"
import { resolve } from "path"
import processLocalDb from "../../src/node/local"
import processDatabaseAsString from "../../src/slim/processDatabaseAsString"

const database = `
0 1 2 3
0 1 2 3
0 1 2 3
0 1 2 3
0 1 2 3
0 1 2
0 1
0 2
`
describe("ProcessDB", () => {
  it("should process local file", () => {
    expect(
      processDatabaseAsString(database)
    ).toMatchSnapshot("processed mushroomset")
  })
})
