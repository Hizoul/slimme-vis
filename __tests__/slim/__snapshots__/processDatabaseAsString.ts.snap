// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProcessDB should process local file: processed mushroomset 1`] = `
Object {
  "0": Array [
    Array [
      1,
      2,
      3,
    ],
    Array [
      1,
      2,
      3,
    ],
    Array [
      1,
      2,
      3,
    ],
    Array [
      1,
      2,
      3,
    ],
    Array [
      1,
      2,
      3,
    ],
    Array [
      1,
      2,
    ],
    Array [
      1,
    ],
    Array [
      2,
    ],
  ],
  "undefined": Array [
    Array [],
    Array [],
  ],
}
`;
